src/VisualStudio/Core/Test/ProjectSystemShim/VisualStudioProjectTests/MetadataToProjectReferenceConversionTests.vb

```diff
@@ -107,5 +109,35 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.ProjectSystemShim
                 Assert.Empty(getReferencingProject().MetadataReferences)
             End Using
         End Sub
+
+        ' This is a test for a potential race between two operations; with 20 iterations on my machine either all would fail
+        ' or one might pass, it seems the race is easy enough to hit without the fix.
+        <WpfTheory>
+        <IterationData(20)>
```

Comment:
><IterationData(20)> [](start = 8, length = 19)

Nit: seems like complicated way to write a for loop. What's the benefit of this attribute vs just a simple for loop in the test?

Reply:
Because that as it turns out has challenges! See https://github.com/dotnet/roslyn/issues/25863 for some further details.

Basically, each time we create a new TestEnvironment we try to create a new MEF container by calling IExportProviderFactory.CreateExportProvider(); this however because of that bug returns the existing MEF container as a singleton for a unit test; this means the second cycle around the loop is reusing the existing Workspace from the MEF composition, which was already Dispose()'ed by the previous loop. Doing the Theory approach ensures they're fully isolated.

@sharwell should probably chime in here and discuss whether we really should be fixing #25863 -- I'll admit I was most surprised when I discovered that.